### PR TITLE
Support non-uniformly spaced data in regression

### PR DESCRIPTION
**Pull request by jat255**
Prior to this change, the statsmodels linear regression calculation assumed that the input table data was uniformly spaced (the "x" column was just a uniformly increasing integer). This resulted in incorrect "slope" calculations on data that was nonuniformly-spaced. This change modifies the function's behavior to map the input `_time` columns to a time difference in nanoseconds, and use that value for the calculation of the regression, meaning its output now matches other tools such as Excel's or Libreoffice Calc's "trend lines" analysis.

I'm a very basic user of Influx and am totally unfamiliar with its development, so I'm not sure how or where to put test cases and will need some direction on that.

@Anaisdg will probably want to take a look

### Checklist

Dear Author :wave:, the following checks should be completed (or explicitly dismissed) before merging.

- [x] ✏️ Write a PR description, regardless of triviality, to include the _value_ of this PR
- [x] 🔗 ~~Reference related issues~~ (did not create one)
- [ ] 🏃 Test cases are included to exercise the new code (I'm not really sure how to do this)

Dear Reviewer(s) :wave:, you are responsible (among others) for ensuring the completeness and quality of the above before approval.

### CODE REVIEW

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux
Comment (sanderson) [inline stdlib/contrib/anaisdg/statsmodels/linearreg.flux:56]: This method assumes that the first timestamp in the first table in the stream of table should be used as the "origin" time for all other tables. I don't think that's a safe assumption. I think a safer assumption would be to use the `_start` value of the row as the origin timestamp. This time represents the query boundary, or if the data is windowed, the window boundary. This would also negate the need to use `findRecord` at the beginning of the function.

```suggestion
            |> map(fn: (r) => ({ r with x : float(v: uint(v: r._time)  - uint(v: r._start)) }))
```

**Review by sanderson — CHANGES_REQUESTED** [review]

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux
Comment (jat255) [reply stdlib/contrib/anaisdg/statsmodels/linearreg.flux]: Good catch, this was a comment left over from a previous version where I did things in seconds

File: stdlib/contrib/anaisdg/statsmodels/linearreg.flux
Comment (jat255) [reply stdlib/contrib/anaisdg/statsmodels/linearreg.flux:56]: I suppose I'm fine with either, but if you use `r._start`, then your regression will be relative to an arbitrary view point (i.e. the beginning of your window/query), rather than the actual data. Playing around with this a bit, if I use a relative query window (i.e. last thirty days), then the slope of my regression changes (very slightly, but still) every time I run the query, since its calculating relative to the query rather than the actual first data point, even though the data contained within my query window has not changed.

> ...first table in the stream of table should be used as the "origin" time for all other tables. I don't think that's a safe assumption.

I use Influx for very simple data, so I don't think I appreciate the consequences of this. I'm willing to defer to your suggestion, but my gut feeling is that it's not "technically" correct, since the regression should be performed on just the data itself, rather than as a function of the viewport.

**Comment by jat255** [comment]
This is stale because I was waiting for further feedback

**Comment by jat255** [comment]
Bump again for feedback from @sanderson  or @Anaisdg

**Pull request closed without merging**
